xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-wikimacro/xwiki-platform-rendering-wikimacro-store/src/main/java/org/xwiki/rendering/wikimacro/internal/DefaultWikiMacroRenderer.java

```diff
@@ -592,6 +592,10 @@ private Block transform(XDOM macroXDOM, boolean async, boolean cached) throws Re
             restoreBindingsOrClean();
         }
 
+        // Update the macro content to take into consideration any change that could have happened to the content
+        // during the macro execution.
+        this.macroContent = macroBinding.getContent();
```

Comment:
Might make more sense to put that in #extractResult since that's where we deal with this kind of things (the result binding for example).

Reply:
Should be fine now